store.go

```diff
@@ -320,6 +334,11 @@ type Store interface {
 	// ApplyDiffFromStagingDirectory uses stagingDirectory to create the diff.
 	ApplyDiffFromStagingDirectory(to, stagingDirectory string, diffOutput *drivers.DriverWithDifferOutput, options *drivers.ApplyDiffWithDifferOpts) error
```

Comment:
Should this be deprecated? I don’t have a strong opinion.

`PutLayer` exists and `CreateLayer` isn’t deprecated, that is one argument against deprecating.

Reply:
yes good idea, let's deprecate it and remove once c/image uses the new API